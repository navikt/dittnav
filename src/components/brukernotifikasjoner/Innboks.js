import React from 'react';
import { arrayOf } from 'prop-types';
import { useIntl } from 'react-intl';
import sikkerhetsContext from '../../hooks/useSikkerhetsnivaa';
import LenkepanelMedIkon from '../common/LenkepanelMedIkon';
import IkonInnboks from '../../assets/IkonInnboks';
import PanelOverskrift from '../common/PanelOverskrift';
import InnloggingsstatusType from '../../types/InnloggingsstatusType';
import InnboksType from '../../types/InnboksType';
import { transformTolokalDatoTid } from '../../utils/datoUtils';
import { lenker } from '../../utils/lenker';
import { listOfActions, listOfComponentNames } from '../../utils/amplitudeUtils';

const Innboks = ({ innbokser, innloggingsstatus }) => {
  const intl = useIntl();

  if (innbokser.length === 0) {
    return null;
  }

  const sikkerhetsnivaa = sikkerhetsContext(innbokser[0], 'innboks', innloggingsstatus);
  const gruppert = innbokser.length > 1;
  const gruppertTekst = intl.formatMessage({ id: 'innboks.flere.meldinger' }, { count: innbokser.length });
  const overskrift = <PanelOverskrift overskrift={gruppert ? gruppertTekst : sikkerhetsnivaa.tekst} type="Element" />;
  const lokalDatoTid = transformTolokalDatoTid(innbokser[0].eventTidspunkt);

  return (
    <LenkepanelMedIkon
      className="innboks-notifikasjon"
      alt="Innboks"
      overskrift={overskrift}
      etikett={gruppert ? null : lokalDatoTid}
      href={gruppert ? lenker.minInnboks.url : sikkerhetsnivaa.lenke}
      amplitudeAction={listOfActions.TrykkPaaBrukernotifikasjon}
      amplitudeComponentName={listOfComponentNames.brukernotifikasjon.InnboksMeldingOppsummering}
    >
      <IkonInnboks />
    </LenkepanelMedIkon>
  );
};

Innboks.propTypes = {
  innbokser: arrayOf(InnboksType),
  innloggingsstatus: InnloggingsstatusType,
};

Innboks.defaultProps = {
  innbokser: null,
  innloggingsstatus: null,
};

export default Innboks;
