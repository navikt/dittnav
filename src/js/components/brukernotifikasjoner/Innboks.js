import React from 'react';
import useSikkerhetsnivaa from '../../hooks/useSikkerhetsnivaa';
import LenkepanelMedIkon from '../common/LenkepanelMedIkon';
import IkonInnboks from '../../../assets/IkonInnboks';
import PanelOverskrift from '../common/PanelOverskrift';
import InnloggingType from '../../types/InnloggingType';
import InnboksType from '../../types/InnboksType';

const Innboks = ({ innboks, innlogging }) => {
  const sikkerhetsnivaa = useSikkerhetsnivaa(innboks, 'innboks', innlogging);
  const overskrift = <PanelOverskrift overskrift={sikkerhetsnivaa.tekst} type="Element" />;

  return (
    <LenkepanelMedIkon
      className="innboks"
      data-ga="Dittnav/Varsel"
      alt="Innboks"
      overskrift={overskrift}
      href={sikkerhetsnivaa.lenke}
    >
      <IkonInnboks />
    </LenkepanelMedIkon>
  );
};

Innboks.propTypes = {
  innboks: InnboksType,
  innlogging: InnloggingType,
};

Innboks.defaultProps = {
  innboks: null,
  innlogging: null,
};

export default Innboks;
