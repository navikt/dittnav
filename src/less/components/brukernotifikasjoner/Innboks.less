.innboks {
  margin-bottom: 1em;
  padding: 0.75em 0.75em !important;;
  display: flex;
  flex-direction: row;
  background-color: #fff8ef !important;
  border: 1px solid @navOransjeLighten60 !important;

  .lenkepanel__innhold {
    padding: 0;
  }

  .lenkepanel__ikon {
    align-self: center;

    svg {
      vertical-align: middle;
    }
  }

  &__tekst {
    align-self: center;
    padding-left: 1em;
  }

  &__knapp {
    align-self: center;
    margin-left: auto;
    padding-right: 0.5em;
  }
}

@media screen and (max-width: @mobile-breakpoint) {
  .innboks {
    padding: 1.025em 1.025em;
    padding-left: 1em;
    padding-top: 1.25em !important;
    padding-bottom: 1.25em !important;


    .lenkepanel__ikon {
      display: none;
    }
  }
}
