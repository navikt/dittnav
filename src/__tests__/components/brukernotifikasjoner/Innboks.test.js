import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import Innboks from 'js/components/brukernotifikasjoner/Innboks';
import wrapIntl from 'js/IntlTestHelper';

test('Innboks with sikkerhetsnivaa 4 and innloggingsnivaa 4', () => {
  const innboks = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Innboks innboks={innboks} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Innboks with sikkerhetsnivaa 4 and innloggingsnivaa 3', () => {
  const innboks = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Innboks innboks={innboks} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Innboks with sikkerhetsnivaa 3 and innloggingsnivaa 3', () => {
  const innboks = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 3,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Innboks innboks={innboks} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Innboks with sikkerhetsnivaa 3 and innloggingsnivaa 4', () => {
  const innboks = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44',
    link: 'https://enNyLenke',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 3,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Innboks innboks={innboks} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});
