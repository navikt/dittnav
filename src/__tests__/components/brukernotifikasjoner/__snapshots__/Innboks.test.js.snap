// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Innboks with sikkerhetsnivaa 3 and innloggingsnivaa 3 1`] = `
<a
  className="lenkepanel innboks-notifikasjon lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#97E6FF"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M36.8182 35.3636H17.1818C15.9785 35.3636 15 34.3851 15 33.1818V21.1818C15 19.9785 15.9785 19 17.1818 19H36.8182C38.0215 19 39 19.9785 39 21.1818V33.1818C39 34.3851 38.0215 35.3636 36.8182 35.3636ZM16.1473 20.8355C16.1107 20.9443 16.0909 21.0608 16.0909 21.1818V33.1818C16.0909 33.7829 16.5796 34.2727 17.1818 34.2727H36.8182C37.4204 34.2727 37.9091 33.7829 37.9091 33.1818V21.1818C37.9091 21.0611 37.8894 20.9449 37.853 20.8362L27.342 29.2449C27.2417 29.3235 27.1206 29.3639 27.0006 29.3639C26.8806 29.3639 26.7595 29.3235 26.6591 29.2449L16.1473 20.8355ZM16.9874 20.1082C17.0505 20.0969 17.1155 20.0909 17.1818 20.0909H36.8182C36.8855 20.0909 36.9514 20.097 37.0154 20.1088L27.0006 28.1202L16.9874 20.1082Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Innboks with sikkerhetsnivaa 3 and innloggingsnivaa 4 1`] = `
<a
  className="lenkepanel innboks-notifikasjon lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#97E6FF"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M36.8182 35.3636H17.1818C15.9785 35.3636 15 34.3851 15 33.1818V21.1818C15 19.9785 15.9785 19 17.1818 19H36.8182C38.0215 19 39 19.9785 39 21.1818V33.1818C39 34.3851 38.0215 35.3636 36.8182 35.3636ZM16.1473 20.8355C16.1107 20.9443 16.0909 21.0608 16.0909 21.1818V33.1818C16.0909 33.7829 16.5796 34.2727 17.1818 34.2727H36.8182C37.4204 34.2727 37.9091 33.7829 37.9091 33.1818V21.1818C37.9091 21.0611 37.8894 20.9449 37.853 20.8362L27.342 29.2449C27.2417 29.3235 27.1206 29.3639 27.0006 29.3639C26.8806 29.3639 26.7595 29.3235 26.6591 29.2449L16.1473 20.8355ZM16.9874 20.1082C17.0505 20.0969 17.1155 20.0909 17.1818 20.0909H36.8182C36.8855 20.0909 36.9514 20.097 37.0154 20.1088L27.0006 28.1202L16.9874 20.1082Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Innboks with sikkerhetsnivaa 4 and innloggingsnivaa 3 1`] = `
<a
  className="lenkepanel innboks-notifikasjon lenkepanel--border"
  href="undefined"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#97E6FF"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M36.8182 35.3636H17.1818C15.9785 35.3636 15 34.3851 15 33.1818V21.1818C15 19.9785 15.9785 19 17.1818 19H36.8182C38.0215 19 39 19.9785 39 21.1818V33.1818C39 34.3851 38.0215 35.3636 36.8182 35.3636ZM16.1473 20.8355C16.1107 20.9443 16.0909 21.0608 16.0909 21.1818V33.1818C16.0909 33.7829 16.5796 34.2727 17.1818 34.2727H36.8182C37.4204 34.2727 37.9091 33.7829 37.9091 33.1818V21.1818C37.9091 21.0611 37.8894 20.9449 37.853 20.8362L27.342 29.2449C27.2417 29.3235 27.1206 29.3639 27.0006 29.3639C26.8806 29.3639 26.7595 29.3235 26.6591 29.2449L16.1473 20.8355ZM16.9874 20.1082C17.0505 20.0969 17.1155 20.0909 17.1818 20.0909H36.8182C36.8855 20.0909 36.9514 20.097 37.0154 20.1088L27.0006 28.1202L16.9874 20.1082Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Du har fått en melding, logg inn med høyere sikkerhetsnivå for å se meldingen.
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Innboks with sikkerhetsnivaa 4 and innloggingsnivaa 4 1`] = `
<a
  className="lenkepanel innboks-notifikasjon lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#97E6FF"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M36.8182 35.3636H17.1818C15.9785 35.3636 15 34.3851 15 33.1818V21.1818C15 19.9785 15.9785 19 17.1818 19H36.8182C38.0215 19 39 19.9785 39 21.1818V33.1818C39 34.3851 38.0215 35.3636 36.8182 35.3636ZM16.1473 20.8355C16.1107 20.9443 16.0909 21.0608 16.0909 21.1818V33.1818C16.0909 33.7829 16.5796 34.2727 17.1818 34.2727H36.8182C37.4204 34.2727 37.9091 33.7829 37.9091 33.1818V21.1818C37.9091 21.0611 37.8894 20.9449 37.853 20.8362L27.342 29.2449C27.2417 29.3235 27.1206 29.3639 27.0006 29.3639C26.8806 29.3639 26.7595 29.3235 26.6591 29.2449L16.1473 20.8355ZM16.9874 20.1082C17.0505 20.0969 17.1155 20.0909 17.1818 20.0909H36.8182C36.8855 20.0909 36.9514 20.097 37.0154 20.1088L27.0006 28.1202L16.9874 20.1082Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Samtalereferat fra telefonsamtale 08.12.2019 kl. 11:44
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;
